test/type/explicit_existential.swift

```diff
@@ -334,12 +334,12 @@ func testEnumAssociatedValue() {
 
 // https://github.com/apple/swift/issues/58920
 typealias Iterator = any IteratorProtocol
-var example: any Iterator = 5 // expected-error{{redundant 'any' has no effect on existential type 'Iterator' (aka 'any IteratorProtocol')}} {{14-18=}} 
+var example: any Iterator = 5 // expected-error{{redundant 'any' in type 'any 'Iterator' (aka 'any IteratorProtocol')'}} {{14-18=}}
 // expected-error@-1{{value of type 'Int' does not conform to specified type 'IteratorProtocol'}}
-var example1: any (any IteratorProtocol) = 5 // expected-error{{redundant 'any' has no effect on existential type 'any IteratorProtocol'}} {{15-19=}}
+var example1: any (any IteratorProtocol) = 5 // expected-error{{redundant 'any' in type 'any 'any IteratorProtocol''}} {{15-19=}}
```

Comment:
Not sure about this, in that case `'any 'any IteratorProtocol''` `any any` looks redundant. If the goal is simplify the message I would suggest something like `redundant 'any' in type  'any IteratorProtocol'` maybe? @hborla WDYT?

Reply:
I think we do want to show both `any`s - the purpose is to illustrate the redundancy, and the fix-it will remove the first `any`. IMO, saying `redundant 'any' in type 'any IteratorProtocol'` makes it sound like `any` isn't required at all and you should just write `IteratorProtocol`.

However, I think we want to show the redundant `any` by creating an `ExistentialType` when the error message is produced rather than hardcoding an extra `any` into the error message to avoid the nested `'`.

To do this, we'll need to get rid of the assertion inside `ExistentialType::get` that fires when an existential is created with a non-constraint type, but I think we should allow such an `ExistentialType` to be created for the purpose of diagnostics.

Reply:
> I think we do want to show both `any`s - the purpose is to illustrate the redundancy, and the fix-it will remove the first `any`. IMO, saying `redundant 'any' in type 'any IteratorProtocol'` makes it sound like `any` isn't required at all and you should just write `IteratorProtocol`.
> 

Ah I see, the idea now. Thank you for clarification =] 

> To do this, we'll need to get rid of the assertion inside `ExistentialType::get` that fires when an existential is created with a non-constraint type, but I think we should allow such an `ExistentialType` to be created for the purpose of diagnostics.

It is also possible just for this purpose to instead of a `Type` make the diagnostic take `StringRef` instead, `"redundant 'any' in type 'any %0'",  (StringRef)` and pass string representation of the type(Type::getString should do that). That way we avoid to have logic to conditionalize assertion on `ExistentialType::get` only for diagnostics or remove the assertion which may be important to keep in a general way. Let me know if that makes sense =]

Reply:
> It is also possible just for this purpose to instead of a Type make the diagnostic take StringRef instead, "redundant 'any' in type 'any %0'", (StringRef) and pass string representation of the type

This is a good thought, but I don't think there's a way to print the underlying type of a type alias via the 'aka' printing this way since that's done through the diagnostic engine.

Reply:
ah yeah, that is unfortunate. I was just trying to think about a way to do that without touching that assertion or creating an additional logic for diagnostics there. But I miss that case, thanks for giving that a thought :)